src/test/java/org/jenkinsci/plugins/android_lint/LintWorkflowTest.java

```diff
@@ -0,0 +1,77 @@
+package org.jenkinsci.plugins.android_lint;
+
+import hudson.FilePath;
+import hudson.model.Result;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import static org.junit.Assert.assertTrue;
+
+public class LintWorkflowTest {
+
+    @Rule
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+
+    /**
+     * Run a workflow job using {@link LintPublisher} and check for success.
+     */
+    @Test
+    public void lintPublisherWorkflowStep() throws Exception {
+        WorkflowJob job = jenkinsRule.jenkins.createProject(WorkflowJob.class, "lintPublisherWorkflowStep");
+        FilePath workspace = jenkinsRule.jenkins.getWorkspaceFor(job);
+        FilePath report = workspace.child("target").child("lint-results.xml");
+        report.copyFrom(getClass().getResourceAsStream("./parser/lint-results_r20.xml"));
+        job.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  step([$class: 'LintPublisher'])\n"
+                        + "}\n", true)
+        );
+        jenkinsRule.assertBuildStatusSuccess(job.scheduleBuild2(0));
+        LintResultAction result = job.getLastBuild().getAction(LintResultAction.class);
+        assertTrue(result.getResult().getAnnotations().size() == 4);
+    }
+
+    /**
+     * Run a workflow job using {@link LintPublisher} with a failing threshold of 0, so the given example file
+     * "/org/jenkinsci/plugins/android_lint/parser/lint-results_r20.xml" will make the build to fail.
+     */
+    @Test
+    public void lintPublisherWorkflowStepSetLimits() throws Exception {
+        WorkflowJob job = jenkinsRule.jenkins.createProject(WorkflowJob.class, "lintPublisherWorkflowStepSetLimits");
+        FilePath workspace = jenkinsRule.jenkins.getWorkspaceFor(job);
+        FilePath report = workspace.child("target").child("lint-results.xml");
+        report.copyFrom(getClass().getResourceAsStream("./parser/lint-results_r20.xml"));
+        job.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  step([$class: 'LintPublisher', pattern: '**/lint-results.xml', failedTotalAll: '0', usePreviousBuildAsReference: false])\n"
+                        + "}\n", true)
+        );
+        jenkinsRule.assertBuildStatus(Result.FAILURE, job.scheduleBuild2(0).get());
+        LintResultAction result = job.getLastBuild().getAction(LintResultAction.class);
+        assertTrue(result.getResult().getAnnotations().size() == 4);
```

Comment:
Thanks. The reason is because `4` isn't a `boolean`, so `assertEquals` doesn't make sense and because using the correct method gives this:

```
java.lang.AssertionError: expected:<4> but was:<0>
        at org.junit.Assert.assertEquals(Assert.java:542)
        at ...(LintWorkflowTest.java:34)
```

Which is a far more helpful error message than:

```
java.lang.AssertionError: null
        at org.junit.Assert.assertTrue(Assert.java:52)
        at ...(LintWorkflowTest.java:34)
```